rbs/prism/CommentsAssociatorPrism.cc

```diff
@@ -652,6 +672,30 @@ void CommentsAssociatorPrism::walkNode(pm_node_t *node) {
             }
             break;
         }
+        case PM_CALL_AND_WRITE_NODE: {
+            walkReceiverWriteNode<pm_call_and_write_node_t>(node, "and_asgn");
+            break;
+        }
+        case PM_CALL_OR_WRITE_NODE: {
+            walkReceiverWriteNode<pm_call_or_write_node_t>(node, "or_asgn");
+            break;
+        }
+        case PM_CALL_OPERATOR_WRITE_NODE: {
+            walkReceiverWriteNode<pm_call_operator_write_node_t>(node, "op_asgn");
+            break;
+        }
+        case PM_INDEX_AND_WRITE_NODE: {
+            walkReceiverWriteNode<pm_index_and_write_node_t>(node, "and_asgn");
+            break;
+        }
+        case PM_INDEX_OR_WRITE_NODE: {
```

Comment:
I feel like this is valid RBS usage that we don't currently support because we don't walk the `ArgumentsNode` in `walkReceiverWriteNode`. We can tackle this in another PR though, we'll also have to add tests too.
```
  arr[
    foo  #: as Integer
  ] ||= "default"
```

Reply:
I added `walkIndexAssignmentNode` which walks the arguments, hopefully that's correct.